rinja_derive/src/heritage.rs

```diff
@@ -91,9 +96,28 @@ impl Context<'_> {
                         let path = config.find_template(import.path, Some(path))?;
                         imports.insert(import.scope, path);
                     }
-                    Node::Extends(_) | Node::Macro(_) | Node::Import(_) if !top => {
-                        return Err(CompileError::no_file_info(
-                            "extends, macro or import blocks not allowed below top level",
+                    Node::Extends(e) if !top => {
```

Comment:
I'm a bit sad about needing to split this in 3 parts (because the `Node` isn't the same, so rust isn't happy). But we got a small improvement for this: the error message is now specific to the node type.

Reply:
I feel the same.  A `name()` on node, optionally with a `top_level_only()`, might reduce the specialization.

The bigger problem I find using Rinja (or Askama) is the lack of context in the error messages for templates. It's nice to know what specific keyword caused the problem but it would be even better if it showed a template line number.

Reply:
It's exactly what this PR adds. Take a look at the error output [here](https://github.com/rinja-rs/rinja/pull/42/files#diff-3de8e2fe64619a6cbe36d7120211f8c3b211198bc3736605eac04d98a03714a0R2-R3). Almost all errors now have context and everything minus a few exceptions. :)

Reply:
Very helpful, thank you!

Reply:
Getting a review from @Kijewski. For now we're two maintainers and we follow the rule to wait for at least one other maintainer's review before merging. So just a bit of patience now. :wink: